src/lib.rs

```diff
@@ -145,12 +146,34 @@ impl AutoCfg {
             return Err(error::from_str("output path is not a writable directory"));
         }
 
+        // Cargo only applies RUSTFLAGS for building TARGET artifact in
+        // cross-compilation environment. Sadly, we don't have a way to detect
+        // when we're building HOST artifact in a cross-compilation environment,
+        // so for now we only apply RUSTFLAGS when cross-compiling an artifact.
+        //
+        // See https://github.com/cuviper/autocfg/pull/10#issuecomment-527575030.
+        let rustflags = env::var("RUSTFLAGS")
+            .ok()
+            .filter(|_| env::var_os("TARGET") != env::var_os("HOST"))
```

Comment:
[`Option::filter`](https://doc.rust-lang.org/std/option/enum.Option.html#method.filter) is only available since 1.27.